tests/v2/validation/rbac/globalrolesv2/namespacedrules_test.go

```diff
@@ -0,0 +1,531 @@
+	ns.validateNSRulesRBACResources(createdUser, namespacedRules)
+
+	log.Info("Verify user can list secrets in the namespace from the namespaced rules")
+	userClient, err := ns.client.AsUser(createdUser)
+	require.NoError(ns.T(), err)
+
+	log.Info("Updating global role by adding multiple namespaced rules.")
+	customNS, err := createProjectAndAddANamespace(ns.client, "ns-readcrtbs")
+	require.NoError(ns.T(), err)
+
+	namespacedRules[customNS] = []rbacv1.PolicyRule{readCRTBsPolicy}
+	updatedGlobalRoleWithMultiNamespacedRules := v3.GlobalRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: createdGlobalRole.Name,
+		},
+		NamespacedRules: namespacedRules,
+	}
+
+	_, err = rbac.UpdateGlobalRole(ns.client, &updatedGlobalRoleWithMultiNamespacedRules)
+	require.NoError(ns.T(), err)
+
+	ns.validateNSRulesRBACResources(createdUser, namespacedRules)
+
+	log.Info("Verify user cannot list secrets in the custom namespace from the namespaced rules for secrets. ")
+	_, err = secrets.ListSecrets(userClient, localcluster, customNS, metav1.ListOptions{})
+	require.Error(ns.T(), err)
+	k8sError.IsForbidden(err)
+
+	log.Info("Verify user can list CRTBS in the custom namespace from the namespaced rules for CRTBs. ")
+	_, err = userClient.WranglerContext.Mgmt.ClusterRoleTemplateBinding().List(customNS, metav1.ListOptions{})
+	require.NoError(ns.T(), err)
+
+	log.Info("Verify user cannot list CRTBS in the globalDataNamespace from the namespaced rules for CRTBs. ")
+	_, err = userClient.WranglerContext.Mgmt.ClusterRoleTemplateBinding().List(globalDataNamespace, metav1.ListOptions{})
+	require.Error(ns.T(), err)
+	k8sError.IsForbidden(err)
+}
+
+func (ns *NamespacedRulesTestSuite) TestDeleteNamespacedRulesFromGlobalRole() {
+
+	subSession := ns.session.NewSession()
+	defer subSession.Cleanup()
+
+	log.Info("Create a global role with namespaced rules.")
+
+	namespacedRules := map[string][]rbacv1.PolicyRule{
+		globalDataNamespace: {
+			readSecretsPolicy,
+		},
+	}
+	createdGlobalRole, err := createGlobalRoleWithNamespacedRules(ns.client, namespacedRules)
+	require.NoError(ns.T(), err)
+
+	log.Info("Create a user with global role standard user and custom global role.")
+	createdUser, err := users.CreateUserWithRole(ns.client, users.UserConfig(), standardUser, createdGlobalRole.Name)
+	require.NoError(ns.T(), err)
+
+	ns.validateNSRulesRBACResources(createdUser, namespacedRules)
+
+	grbOwner, err := getGlobalRoleBindingForUser(ns.client, createdUser.ID)
+	require.NoError(ns.T(), err)
+	nameSelector := fmt.Sprintf("metadata.name=%s-%s", grbOwner, namespace)
+
+	deleteNamespacedRules := v3.GlobalRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: createdGlobalRole.Name,
+		},
+		NamespacedRules: map[string][]rbacv1.PolicyRule{},
+	}
+
+	_, err = rbac.UpdateGlobalRole(ns.client, &deleteNamespacedRules)
+	require.NoError(ns.T(), err)
+
+	rbs, err := rbac.ListRoleBindings(ns.client, localcluster, namespace, metav1.ListOptions{FieldSelector: nameSelector})
```

Comment:
Can we add a step before this to verify that secrets cannot be listed after deleting the NamespacedRules?